sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/FieldBuilder.java

```diff
@@ -0,0 +1,154 @@
+// Licensed under the MIT License.
+
+package com.azure.search.documents;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.search.annotation.FieldIgnore;
+import com.azure.search.annotation.FieldProperty;
+import com.azure.search.annotation.SearchableFieldProperty;
+import com.azure.search.annotation.SimpleFieldProperty;
+import com.azure.search.documents.models.AnalyzerName;
+import com.azure.search.documents.models.DataType;
+import com.azure.search.documents.models.Field;
+import com.azure.search.documents.models.GeoPoint;
+
+import java.lang.annotation.Annotation;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Helper to build search field
+ */
+public class FieldBuilder {
+    /**
+     * Creates a collection of <see cref="Field"/> objects corresponding to
+     * the properties of the type supplied.
+     *
+     * @param modelClass The class type for which fields will be created, based on its properties.
+     * @return A collection of fields.
+     */
+    public static <T> List<Field> build(Class<T> modelClass) {
+        ClientLogger logger = new ClientLogger(FieldBuilder.class);
+        java.lang.reflect.Field[] fields = modelClass.getFields();
+        List<Field> searchFieldList = new ArrayList<>();
+        for (java.lang.reflect.Field field: fields) {
+            Annotation[] annotations = field.getAnnotations();
+            for (Annotation annotation : annotations) {
+                if (annotation instanceof FieldIgnore) {
+                    continue;
+                }
+                Field searchField = new Field();
+                searchField.setName(field.getName());
+                searchField.setType(covertToDataType(field, logger));
+                if (annotation instanceof SimpleFieldProperty) {
+                    buildSimpleField(searchField, (SimpleFieldProperty) annotation);
+                    searchFieldList.add(searchField);
+                } else if (annotation instanceof SearchableFieldProperty) {
+                    buildSearchableField(searchField, (SearchableFieldProperty) annotation);
+                    searchFieldList.add(searchField);
+                } else if (annotation instanceof FieldProperty) {
+                    buildField(searchField, (FieldProperty) annotation);
+                    searchFieldList.add(searchField);
+                }
+                logger.info("The property {} is not a search field.", field.getName());
+            }
+        }
+        return searchFieldList;
+    }
+
+    private static void buildSimpleField(Field searchField, SimpleFieldProperty annotation) {
+        searchField.setSearchable(true);
+        searchField.setSortable(true);
+        searchField.setFilterable(true);
+        searchField.setFacetable(true);
+        searchField.setKey(annotation.isKey());
+        if (!"null".equals(annotation.analyzer())) {
```

Comment:
Mentioned above, but just to reiterate here (and applies below): analyzers, synonymMaps, etc., are only settable on searchable fields and shouldn't be here.